namesilo.php

```diff
@@ -366,15 +366,23 @@ public function addService(
                         if (reset($this->Input->errors()['errors']) === $error) {
                             // unset the errors since we are working around it
                             $this->Input->setErrors([]);
+
                             // set the registration length to 1 year and save the remainder for an extension
                             $total_years = $fields['years'];
                             $fields['years'] = 1;
                             $response = $domains->create($fields);
                             $this->processResponse($api, $response);
+
                             // now extend the remainder of the years
                             $fields['years'] = $total_years - 1;
                             $response = $domains->renew($fields);
                             $this->processResponse($api, $response);
+
+                            // set privacy status
+                            if (((bool) $vars['private'] || isset($vars['configoptions']['id_protection'])) && $this->supportsIdProtection()) {
```

Comment:
I believe the solution is actually a bit simpler than this.  On line 263, a list of fields to send to the api is gathered into $input_fields.  This list used to include the 'private' field, but when we switched to using the config option for this, it got erroneously removed from the list.

I believe it is sufficient to change line 263 from:

```
        $input_fields = array_merge(
            Configure::get('Namesilo.domain_fields'),
            (array) Configure::get('Namesilo.domain_fields' . $tld),
            (array) Configure::get('Namesilo.nameserver_fields'),
            (array) Configure::get('Namesilo.transfer_fields'),
            ['years' => true, 'transfer' => isset($vars['transfer']) ? $vars['transfer'] : 1]
        );
```
To
```
        $input_fields = array_merge(
            Configure::get('Namesilo.domain_fields'),
            (array) Configure::get('Namesilo.domain_fields' . $tld),
            (array) Configure::get('Namesilo.nameserver_fields'),
            (array) Configure::get('Namesilo.transfer_fields'),
            ['years' => true, 'transfer' => isset($vars['transfer']) ? $vars['transfer'] : 1, 'private' => 0]
        );
```